amethyst_tiles/src/map.rs

```diff
@@ -290,13 +301,36 @@ fn create_transform(map_dimensions: &Vector3<u32>, tile_dimensions: &Vector3<u32
 }
 
 #[allow(clippy::cast_precision_loss)]
-fn to_world(transform: &Matrix4<f32>, coord: &Point3<u32>) -> Vector3<f32> {
+fn to_world(
+    transform: &Matrix4<f32>,
+    coord: &Point3<u32>,
+    map_transform: Option<&Transform>,
+) -> Vector3<f32> {
     let coord_f = Point3::new(coord.x as f32, -1.0 * coord.y as f32, coord.z as f32);
-    transform.transform_point(&coord_f).coords
+    if let Some(map_trans) = map_transform {
+        map_trans
+            .global_view_matrix()
```

Comment:
This should use `Transform::global_matrix()`instead of `global_view_matrix().inverse()`, as the view_matrix is already the inverse of the global_matrix, so we are doing 2 inversions here for no reason.

Reply:
Done